packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -254,6 +256,11 @@ export class CdkToolkit {
           }
 
           const confirmed = await promptly.confirm('Do you wish to deploy these changes (y/n)?');
+
+          setLogLock(false);
```

Comment:
This needs to be in a `try/finally` or the cleanup might not happen in case of an exception.

You could also make:

```ts
async function withCorkedLogging<A>(block: () => Promise<A>): Promise<A> {
  corkLogging();
  try {
    return block();
  } finally {
    uncorkLogging();
  }
}
```